### ISSUE TITLE
CSV supersize me

### ISSUE THREAD

**Issue by susanodd** [Signbank/Global-signbank#558]
Regarding the Big CSV file of @ocrasborn:
So all glosses to create are displayed in a giant glosses to create table.
But upon pressing the button to make changes:

`Request body exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE.`

**Comment by Woseseltops** [comment]
Idea for last resort: create a local offline tool to split up CSVs into smaller ones.

**Comment by susanodd** [comment]
On my local machine, the supersized csv file was successfully imported, and 16,928 glosses were created. Based on my activities app (the time I started walking home) and the time on the signbank.db file, it took circa 20 minutes to create the glosses. Since I was waiting a bit to see if it would crash, say it took 30 minutes.

**Comment by susanodd** [comment]
So we know the import code works, but it neglected to report two errors in @ocrasborn's csv file.
And I modified the base.py setting with:
`DATA_UPLOAD_MAX_MEMORY_SIZE = None`

**Comment by susanodd** [comment]
I pushed some changes to github that should solve the problem.

To locate the problems, I put the two erroneous rows of @ocrasborn's supersized csv into a separate file and was able to get the system to hang with just those two rows of data. I'll use that data as a new test.

I rewrote some of the code to move things outside of the loops, added an extra break in case errors are found.
(The intention of the code is to give as much feedback as possible, but when the file is huge, this seems to be a black hole. The already seen annotation id glosses were not being kept properly if there had been errors in them. So now the code will give feedback sooner rather than trying to keep processing.)
There was also a sloppy query to look up glosses with the same annotation for the language. But that had been implemented in a totally roundabout way and was generating tons of queries inside a loop.
That has been changed to a single query.
Plus the maximum upload memory setting has no limit.

**Comment by susanodd** [comment]
Okay, I noticed that @vanlummelhuizen put the revised code on the server yesterday.
Because the mega file worked on my server, I tried to import the huge file on the live server.
But it locked the database before any feedback was shown.
This is very strange because it works on my server.
The only difference is that I have DEBUG set to False, whereas it's True on the live server.

**Comment by Woseseltops** [comment]
Progress together with @susanodd today. We logged every line as it was looping through Onno's huge CSV file. We noticed:
* It seems to work; we see every single line being processed.
* There are a lot of database queries during the importing process, and every now and then we see a database locked error (produced by us? produced by someone else) being logged. Not sure if this is a problem.
* After the last line, we get:

```Fri Mar  8 14:57:56 2019 - uwsgi_response_write_body_do(): Broken pipe [core/writer.c line 331] during POST /signs/import_csv_create/ (131.174.132.138)
```

This most likely indicates a timeout, so after Django is done and wants to send back its results, Apache tells it it's too late already. A simple solution would be to make the timeout setting in Apache longer?
